service/tablet_allocator.cc

```diff
@@ -1449,6 +1449,18 @@ class load_balancer {
                     });
                 }
 
+                // Node which is draining is either being decommissioned or removed.
+                // If involved node is excluded, co-locating migration will surely fail, so it's pointless.
+                // We should wait until the node is removed.
+                // Also, it can fail the removenode request, as failure of this migration is interpreted as
+                // draining failure.
+                // In case of decommission, draining is more important than co-location, so postponing is good.
+                if (nodes.at(dst.host).drained || nodes.at(src.host).drained) {
```

Comment:
Why do we even consider a node that is been drained as a destination for migration?